pkg/gcp/secret.go

```diff
@@ -0,0 +1,92 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const secretFormat = "projects/%s/secrets/%s"
+
+func makeSecretURL(projectID, secretName string) string {
+	return fmt.Sprintf(secretFormat, projectID, secretName)
+}
+
+// SecretClient is GCP compute client with go client
+type SecretClient struct {
+	client    *secretmanager.Client
+	projectID string
+	ctx       context.Context
+}
+
+// NewSecretClient returns SecretClient
+func NewSecretClient(
+	ctx context.Context,
+	projectID string,
+) (*SecretClient, error) {
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &SecretClient{
+		client:    client,
+		projectID: projectID,
+		ctx:       ctx,
+	}, nil
+}
+
+func (c *SecretClient) GetSecret(name string) (*secretmanagerpb.Secret, error) {
+	getSecretReq := secretmanagerpb.GetSecretRequest{Name: makeSecretURL(c.projectID, name)}
+	return c.client.GetSecret(c.ctx, &getSecretReq)
+}
+
+func (c *SecretClient) Exists(name string) (bool, error) {
+	_, err := c.GetSecret(name)
+	if err == nil {
+		return true, nil
+	}
+	if code := status.Code(err); code == codes.NotFound {
+		return false, nil
+	} else {
+		return false, err
+	}
+}
+
+func (c *SecretClient) CreateSecretFromData(expireTime time.Time, secretName string, data []byte) error {
+	secretPath := makeSecretURL(c.projectID, secretName)
+	createSecretReq := &secretmanagerpb.CreateSecretRequest{
+		Parent:   "projects/" + c.projectID,
+		SecretId: secretName,
+		Secret: &secretmanagerpb.Secret{
+			Replication: &secretmanagerpb.Replication{
+				Replication: &secretmanagerpb.Replication_Automatic_{
+					Automatic: &secretmanagerpb.Replication_Automatic{},
+				},
+			},
+			Expiration: &secretmanagerpb.Secret_ExpireTime{ExpireTime: timestamppb.New(expireTime)},
+		},
+	}
+	_, err := c.client.CreateSecret(c.ctx, createSecretReq)
+	if err != nil {
+		if stat, ok := status.FromError(err); ok && stat.Code() == codes.AlreadyExists {
+			return fmt.Errorf("secret %s is already exists", secretPath)
+		} else {
+			return err
+		}
+	}
+
+	addSecretVersionReq := &secretmanagerpb.AddSecretVersionRequest{
+		Parent: secretPath,
```

Comment:
`secretPath` is practically used only here. So it is enough to call the function directly.

```suggestion
		Parent: makeSecretURL(c.projectID, secretName),
```